core/client/db/src/lib.rs

```diff
@@ -448,7 +492,7 @@ impl<Block: BlockT> HeaderMetadata<Block> for BlockchainDb<Block> {
 /// Database transaction
 pub struct BlockImportOperation<Block: BlockT, H: Hasher> {
 	old_state: CachingState<Blake2Hasher, RefTrackingState<Block>, Block>,
-	db_updates: PrefixedMemoryDB<H>,
+	db_updates: (PrefixedMemoryDB<H>, InMemoryKvBackend),
```

Comment:
maybe it is better to avoid tuple here ? you could do trie_db_update and kv_db_update.
or maybe create a new struct ?

Reply:
Yes, that is a bit of refactoring, but those key value pr should try to get as decoupled as possible from the trie state (the opposite of what I did).
That would be a big but probably needed refactoring, I am switching those PR state to in_progress thus.